addons/website_sale/controllers/main.py

```diff
@@ -835,8 +835,7 @@ def address(self, **kw):
             else:
                 partner_id = self._checkout_form_save(mode, post, kw)
                 if mode[1] == 'billing':
-                    order.partner_id = partner_id
-                    order.with_context(not_self_saleperson=True).onchange_partner_id()
+                    order.with_context(not_self_saleperson=True).partner_id = partner_id
```

Comment:
Context change can be removed, the key was only used in sale (and you removed it).
(this logic is strange anyway)